### DISCUSSION TITLE
Detecting a curve in a scanned image

### DISCUSSION THREAD

**Discussion by jumpjack** [ImageMagick/ImageMagick#4718] [Digital Image Processing]
I want to extract the curve from chart below; I tried overwriting the grid by a program-generated white grid, but grid in the image is too much irregular, it does not work.
So I would like to try a different approach: detecting the curve rather than removing the grid, as the curve is thicker than (almost) any other line in the image.
Can I do it with IM?
![chart-test-2](https://user-images.githubusercontent.com/1620953/149654785-206410b8-63a5-4fe4-ace0-cba7f3951834.png)

**Comment by snibgo** [comment]
That is a difficult image. For example: what processing will remove the arrows from the annotations but leave the curve?

If this is a one-off, the obvious solution is to use an image editor such as Gimp. Create a new transparent layer. In that layer, draw a line by hand that traces the curve. Export that layer to a file. This will be accurate, and will take only a few minutes.

If you have many such graphs, an automated solution might be developed. One approach: greatly increase contrast to reduce the "thin" lines to one pixel wide. Then use morphology to remove those one-pixel-wide lines. The curve remains, but it is broken by gaps.

**Reply by jumpjack** [reply]
I have a dozen of such images...
Yes, I could redraw by hand the curve in any plot digitizer, but the point is not to touch the curve to avoid introducing errors.
I don't mind arrows, labels and comments, I can manually erase them.
